Gems/Terrain/Code/Source/TerrainRenderer/TerrainFeatureProcessor.cpp

```diff
@@ -68,76 +65,76 @@ namespace Terrain
         EnableSceneNotification();
     }
 
+    void TerrainFeatureProcessor::ConfigurePipelineState(ShaderState& shaderState, bool assertOnFail)
+    {   
+        bool success = GetParentScene()->ConfigurePipelineState(shaderState.m_shader->GetDrawListTag(), shaderState.m_pipelineStateDescriptor);
+        AZ_Assert(success || !assertOnFail, "Couldn't configure the pipeline state.");
+        if (success)
+        {
+            shaderState.m_pipelineState = shaderState.m_shader->AcquirePipelineState(shaderState.m_pipelineStateDescriptor);
+            AZ_Assert(shaderState.m_pipelineState, "Failed to acquire default pipeline state.");
```

Comment:
Should this check assertOnFail too?

Reply:
I don't think so - the first operation, ConfigurePipelineState(), is expected to possibly fail if the render pipeline hasn't been added yet. m_shader->AcquirePipelineState shouldn't fail though so this function doesn't give the option to ignore it.